### PR TITLE
replace minus to dash

### PR DESCRIPTION
**Pull request by xp44mm**

### CODE REVIEW

**Comment by chenxsan** [comment]
Can you share your reason/findings to update them? As I'm not familiar with those punctuation marks in English.

**Comment by xp44mm** [comment]
Please search for **em dash**, which may be helpful to you.

**Comment by chenxsan** [comment]
Thank you for your patience here, I did some research and found that `--` is ok as per https://en.wikipedia.org/wiki/Double_hyphen:

> The double hyphen is also not to be confused with two consecutive hyphens (--), which are often used to represent an em dash — or en dash –; that practice arose due to the limitations of typewriter character sets which did not have both hyphen and dash

**Comment by xp44mm** [comment]
The dash is more formal, and the double minus represents the command argument.

**Comment by chenxsan** [comment]
The point is "typewriter character sets which did not have both hyphen and dash". For instance, I don't know how to input em dash with my keyboard but I do know how to input two consecutive hyphens. I want to keep it friendly for contributors here as `--` won't hurt the experience of end users. And when `--` is used as command argument, it's usually wrapper in `<code></code>` which is quite easy to speculate.

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ internal-links.tap
 stats.json
 printable.md
 repositories/*.json
+/.vs
```
Comment (montogeek) [inline .gitignore:16]: Please don't add your specific setup configuration to the project.

File: src/content/api/index.md [hunk 2]
```diff
@@ -24,8 +24,8 @@ file and a few flags (e.g. `--env`).
 ## Module
 
 When processing modules with webpack, it is important to understand the
-different module syntaxes -- specifically the [methods](/api/module-methods)
-and [variables](/api/module-variables) -- that are supported.
+different module syntaxes — specifically the [methods](/api/module-methods)
+and [variables](/api/module-variables) — that are supported.
```
Comment (chenxsan) [inline src/content/api/index.md:28]:
```suggestion
different module syntaxes – specifically the [methods](/api/module-methods)
and [variables](/api/module-variables) – that are supported.
```

File: src/content/concepts/why-webpack.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ When Node.js was released a new era started, and it came with new challenges. No
 CommonJS came out and introduced `require`, which allows you to load and use a module in the current file. This solved scope issues out of the box by importing each module as it was needed.
 
 
-## npm + Node.js + modules -- mass distribution
+## npm + Node.js + modules — mass distribution
```
Comment (chenxsan) [inline src/content/concepts/why-webpack.md:36]:
```suggestion
## npm + Node.js + modules – mass distribution
```

File: src/content/configuration/other-options.md [hunk 4]
```diff
@@ -387,7 +387,7 @@ T> Combine with `parallelism: 1` for better results.
 
 `string`
 
-Use this option to generate a JSON file containing webpack "records" -- pieces of data used to store module identifiers across multiple builds. You can use this file to track how modules change between builds. To generate one, simply specify a location:
+Use this option to generate a JSON file containing webpack "records" — pieces of data used to store module identifiers across multiple builds. You can use this file to track how modules change between builds. To generate one, simply specify a location:
```
Comment (chenxsan) [inline src/content/configuration/other-options.md:390]:
```suggestion
Use this option to generate a JSON file containing webpack "records" – pieces of data used to store module identifiers across multiple builds. You can use this file to track how modules change between builds. To generate one, simply specify a location:
```

File: src/content/contribute/index.md [hunk 5]
```diff
@@ -70,7 +70,7 @@ You can also encourage your developers to contribute to the ecosystem by open-so
 
 ### Anyone Else
 
-To anyone else who is interested in helping our mission -- e.g. venture capitalists, government entities, digital agencies, etc. -- we would love for you to work with us, one of the top npm packages, to improve your product! Please don't hesitate to reach out with questions.
+To anyone else who is interested in helping our mission — e.g. venture capitalists, government entities, digital agencies, etc. — we would love for you to work with us, one of the top npm packages, to improve your product! Please don't hesitate to reach out with questions.
```
Comment (chenxsan) [inline src/content/contribute/index.md:73]:
```suggestion
To anyone else who is interested in helping our mission – e.g. venture capitalists, government entities, digital agencies, etc. – we would love for you to work with us, one of the top npm packages, to improve your product! Please don't hesitate to reach out with questions.
```

File: src/content/contribute/writing-a-loader.md [hunk 6]
```diff
@@ -68,7 +68,7 @@ T> You can use [`webpack-defaults` package](https://github.com/webpack-contrib/w
 
 ## Simple Usage
 
-When a single loader is applied to the resource, the loader is called with only one parameter -- a string containing the content of the resource file.
+When a single loader is applied to the resource, the loader is called with only one parameter — a string containing the content of the resource file.
```
Comment (chenxsan) [inline src/content/contribute/writing-a-loader.md:71]:
```suggestion
When a single loader is applied to the resource, the loader is called with only one parameter – a string containing the content of the resource file.
```

File: src/content/contribute/writing-a-loader.md [hunk 7]
```diff
@@ -77,7 +77,7 @@ The loader is expected to give back one or two values. The first value is a resu
 
 ## Complex Usage
 
-When multiple loaders are chained, it is important to remember that they are executed in reverse order -- either right to left or bottom to top depending on array format.
+When multiple loaders are chained, it is important to remember that they are executed in reverse order — either right to left or bottom to top depending on array format.
```
Comment (chenxsan) [inline src/content/contribute/writing-a-loader.md:80]:
```suggestion
When multiple loaders are chained, it is important to remember that they are executed in reverse order – either right to left or bottom to top depending on array format.
```

File: src/content/guides/asset-management.md [hunk 8]
```diff
@@ -161,7 +161,7 @@ webpack 5.4.0 compiled successfully in 2900 ms
 
 Open up `dist/index.html` in your browser again and you should see that `Hello webpack` is now styled in red. To see what webpack did, inspect the page (don't view the page source, as it won't show you the result, because the `<style>` tag is dynamically created by JavaScript) and look at the page's head tags. It should contain the style block that we imported in `index.js`.
 
-Note that you can, and in most cases should, [minimize css](/plugins/mini-css-extract-plugin/#minimizing-for-production) for better load times in production. On top of that, loaders exist for pretty much any flavor of CSS you can think of -- [postcss](/loaders/postcss-loader), [sass](/loaders/sass-loader), and [less](/loaders/less-loader) to name a few.
+Note that you can, and in most cases should, [minimize css](/plugins/mini-css-extract-plugin/#minimizing-for-production) for better load times in production. On top of that, loaders exist for pretty much any flavor of CSS you can think of — [postcss](/loaders/postcss-loader), [sass](/loaders/sass-loader), and [less](/loaders/less-loader) to name a few.
```
Comment (chenxsan) [inline src/content/guides/asset-management.md:164]:
```suggestion
Note that you can, and in most cases should, [minimize css](/plugins/mini-css-extract-plugin/#minimizing-for-production) for better load times in production. On top of that, loaders exist for pretty much any flavor of CSS you can think of – [postcss](/loaders/postcss-loader), [sass](/loaders/sass-loader), and [less](/loaders/less-loader) to name a few.
```

File: src/content/guides/lazy-loading.md [hunk 9]
```diff
@@ -22,7 +22,7 @@ Lazy, or "on demand", loading is a great way to optimize your site or applicatio
 
 ## Example
 
-Let's take the example from [Code Splitting](/guides/code-splitting/#dynamic-imports) and tweak it a bit to demonstrate this concept even more. The code there does cause a separate chunk, `lodash.bundle.js`, to be generated and technically "lazy-loads" it as soon as the script is run. The trouble is that no user interaction is required to load the bundle -- meaning that every time the page is loaded, the request will fire. This doesn't help us too much and will impact performance negatively.
+Let's take the example from [Code Splitting](/guides/code-splitting/#dynamic-imports) and tweak it a bit to demonstrate this concept even more. The code there does cause a separate chunk, `lodash.bundle.js`, to be generated and technically "lazy-loads" it as soon as the script is run. The trouble is that no user interaction is required to load the bundle — meaning that every time the page is loaded, the request will fire. This doesn't help us too much and will impact performance negatively.
```
Comment (chenxsan) [inline src/content/guides/lazy-loading.md:25]:
```suggestion
Let's take the example from [Code Splitting](/guides/code-splitting/#dynamic-imports) and tweak it a bit to demonstrate this concept even more. The code there does cause a separate chunk, `lodash.bundle.js`, to be generated and technically "lazy-loads" it as soon as the script is run. The trouble is that no user interaction is required to load the bundle – meaning that every time the page is loaded, the request will fire. This doesn't help us too much and will impact performance negatively.
```

**Review by chenxsan — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**